src/EditorFeatures/Core/FindUsages/FindUsagesHelpers.cs

```diff
@@ -107,6 +107,11 @@ public static async Task<(ISymbol symbol, Project project, ImmutableArray<ISymbo
                     }
                 }
 
+                if (!symbol.IsInterfaceType() && !symbol.IsAbstract)
+                {
+                    return implementationsAndOverrides.Concat(symbol).ToImmutableArray();
```

Comment:
@AlekseyTs can we just add symbol to the implementationsAndOverrides hash set in the new condition and let existing
eturn implementationsAndOverrides.ToImmutableArray(); handle the return?

Reply:
what does this condition actually mean? if symbol is implementableMember + not abstract then it means it is default member?

Reply:
seems good extension method to have if asking whether symbol is interface default implementation or not is not exposed by compiler API.

Reply:
> what does this condition actually mean? if symbol is implementableMember + not abstract then it means it is default member?

It means the member has implementation.

---
In reply to: [269287568](https://github.com/dotnet/roslyn/pull/34433#discussion_r269287568) [](ancestors = 269287568)

Reply:
> seems good extension method to have if asking whether symbol is interface default implementation or not is not exposed by compiler API.

I do not think any new API is needed, ```IsAbstract``` is sufficient.

---
In reply to: [269288400](https://github.com/dotnet/roslyn/pull/34433#discussion_r269288400) [](ancestors = 269288400)

Reply:
> can we just add symbol to the implementationsAndOverrides hash set in the new condition and let existing
eturn implementationsAndOverrides.ToImmutableArray(); handle the return?

I think we can. Do you prefer that?


---
In reply to: [269285650](https://github.com/dotnet/roslyn/pull/34433#discussion_r269285650) [](ancestors = 269285650)

Reply:
yep. since it makes pattern to same so, no need to stop and think about special case such as does it guarantee that there are no duplicates and etc.

Reply:
so if containing type is interface and if member is not abstract then it means it is default implementation?

Reply:
> so if containing type is interface and if member is not abstract then it means it is default implementation?

It means the member has implementation, it is not abstract.

---
In reply to: [269301813](https://github.com/dotnet/roslyn/pull/34433#discussion_r269301813) [](ancestors = 269301813)